FilePersistence/src/version_control/Diff3Parse.cpp

```diff
@@ -162,6 +161,48 @@ QList<Model::NodeIdType> Diff3Parse::longestCommonSubsequence(const QList<Model:
 	return lcs;
 }
 
+QList<Model::NodeIdType> Diff3Parse::longestCommonSubsequence3(const QList<Model::NodeIdType> listA,
+																				const QList<Model::NodeIdType> listB,
+																				  const  QList<Model::NodeIdType> listC)
+{
+	int m = listA.size() + 1;
+	int n = listB.size() + 1;
+	int o = listC.size() + 1;
+
+	int*** lcsLength = new int**[m];
+	for (int i = 0; i < m; ++i)
+	{
+		lcsLength[i] = new int*[n];
+		for (int j = 0; j < n; ++j)
+			lcsLength[i][j] = new int[o];
+	}
+
+	for (int i = 0; i < m; ++i)
+		for (int j = 0; j < n; ++j)
+			for (int k = 0; k < o; ++k)
```

Comment:
Instead of having three brand new loops, let's just insert this last loop, in the loops above -> initialize after after it was constructed.